Controller/ExceptionController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the DunglasApiBundle package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Dunglas\ApiBundle\Controller;
+
+use Dunglas\ApiBundle\JsonLd\Response;
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Log\DebugLoggerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Render a normalized exception for a given {@see \Symfony\Component\Debug\Exception\FlattenException}.
+ *
+ * @author Baptiste Meyer <baptiste.meyer@gmail.com>
+ */
+class ExceptionController
+{
+    /**
+     * @var NormalizerInterface
+     */
+    private $normalizer;
+
+    /**
+     * @param NormalizerInterface $normalizer
+     */
+    public function __construct(NormalizerInterface $normalizer)
+    {
+        $this->normalizer = $normalizer;
+    }
+
+    /**
+     * Converts a {@see \Symfony\Component\Debug\Exception\FlattenException}
+     * to a {@see \Dunglas\ApiBundle\JsonLd\Response}.
+     *
+     * @param Request                   $request
+     * @param FlattenException          $exception
+     * @param DebugLoggerInterface|null $logger
+     *
+     * @return Response
+     */
+    public function showAction(Request $request, FlattenException $exception, DebugLoggerInterface $logger = null)
+    {
+        if (
+            'Dunglas\ApiBundle\Exception\DeserializationException' === $exception->getClass()
+            || is_subclass_of($exception->getClass(), 'Dunglas\ApiBundle\Exception\DeserializationException')
+        ) {
+            $exception->setStatusCode(Response::HTTP_BAD_REQUEST);
+        }
+
+        return new Response(
+            $this->normalizer->normalize($exception, 'hydra-error'),
```

Comment:
@dunglas: IMO it's simpler to define a new configuration option for the `api-bundle`: `exception_controller`. By default, it would use the new `Dunglas\ApiBundle\Controller\ExceptionController`.

Then, we need to replace the class used by `twig.exception_listener` to our own `ExceptionListener`, which would be in charge to change the `_controller` attribute:

```php
protected function duplicateRequest(\Exception $exception, Request $request)
{
    $newRequest = parent::duplicateRequest($exception, $request);
    if ($request->attributes->has('_resource')) {
          $newRequest->attributes->set('_controller', $this->apiExceptionController)
    }

    return $newRequest; 
}
```

But indeed, it might cause issues if someone already redefines its own ExceptionListener.